### PR TITLE
map

### PR DESCRIPTION
**Pull request by AayushK123**
Added apriltag subsystem to robot container and put robot position on shuffleboard map.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/AprilTagPosition.java [hunk 1]
```diff
@@ -95,7 +103,7 @@ private RobotPosition getRightRobotPosition() {
     positionRight.rotation = yRotationRightCam.getDouble(0);
     positionRight.fieldXPosition = fieldXCoordinateRight.getDouble(0);
     positionRight.fieldYPosition = fieldYCoordinateRight.getDouble(0);
-    positionRight.fieldRotation = fieldRotationRight.getDouble(0);
+    positionRight.fieldRotation = fieldRotationRight.getDouble(3.14);
```
Comment (bengold22) [inline src/main/java/frc/robot/subsystems/AprilTagPosition.java]: We need to undo this constant and put it back to 0

File: src/main/java/frc/robot/subsystems/AprilTagPosition.java [hunk 2]
```diff
@@ -73,7 +80,8 @@ public Pose2d getTagRelativePose2d() {
   public Pose2d getFieldRelativePose2d() {
     RobotPosition position = getRobotPosition();
     if (position != null) {
-      return new Pose2d(position.fieldXPosition, position.fieldYPosition, new Rotation2d(position.fieldRotation));
+      Pose2d pose2d = new Pose2d(position.fieldXPosition, position.fieldYPosition, new Rotation2d(position.fieldRotation));
```
Comment (bengold22) [inline src/main/java/frc/robot/subsystems/AprilTagPosition.java]: Put this back to one line (where it returns the Pose2d object directly)

File: src/main/java/frc/robot/subsystems/AprilTagPosition.java [hunk 3]
```diff
@@ -119,4 +127,20 @@ else if (isLeftCameraDetectingTag() && isRightCameraDetectingTag()) {
     }
     return null;
   }
+
+  @Override
+  public void periodic() {
+    RobotPosition position = getRobotPosition();
```
Comment (bengold22) [inline src/main/java/frc/robot/subsystems/AprilTagPosition.java]: Lets wrap all this (everything in periodic) in:

if (Constants.DEBUG == true) { 
}

**Review by bengold22 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/AprilTagPosition.java [hunk 4]
```diff
@@ -119,4 +127,20 @@ else if (isLeftCameraDetectingTag() && isRightCameraDetectingTag()) {
...
+    if (position != null) {
+      SmartShuffleboard.put("apriltag", "fieldX", position.fieldXPosition);
+      SmartShuffleboard.put("apriltag", "fieldY", position.fieldYPosition);
+      SmartShuffleboard.put("apriltag", "rotation", position.fieldRotation);
+      SmartShuffleboard.put("apriltag", "tagRotation", position.rotation);
+      // robotField.setRobotPose(getFieldRelativePose2d());
+      Pose2d robootpose = getFieldRelativePose2d();
+      robotField.setRobotPose(robootpose.getX(), robootpose.getY(), robootpose.getRotation());
```
Comment (bengold22) [inline src/main/java/frc/robot/subsystems/AprilTagPosition.java]: Lets test putting this back to robotField.setRobotPose(getFieldRelativePose2d())

**Review by bengold22 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/AprilTagPosition.java (hunk 1 above)
Comment (jksupabuddies) [reply src/main/java/frc/robot/subsystems/AprilTagPosition.java]: fixed

File: src/main/java/frc/robot/subsystems/AprilTagPosition.java (hunk 2 above)
Comment (jksupabuddies) [reply src/main/java/frc/robot/subsystems/AprilTagPosition.java]: fixed

File: src/main/java/frc/robot/subsystems/AprilTagPosition.java (hunk 3 above)
Comment (jksupabuddies) [reply src/main/java/frc/robot/subsystems/AprilTagPosition.java]: added

File: src/main/java/frc/robot/subsystems/AprilTagPosition.java (hunk 4 above)
Comment (jksupabuddies) [reply src/main/java/frc/robot/subsystems/AprilTagPosition.java]: fixed

**Review by bengold22 — APPROVED** [review]

**Pull request merged**